deepchecks/checks/distribution/whole_dataset_drift.py

```diff
@@ -0,0 +1,234 @@
+                Random seed for the check.
+        """
+
+    def __init__(
+        self,
+        n_top_features: int = 3,
+        min_feature_importance: float = 0.05,
+        max_num_categories: int = 10,
+        sample_size: int = 10000,
+        random_state: int = 0,
+    ):
+        super().__init__()
+
+        self._cat_features = None
+        self.n_top_features = n_top_features
+        self.min_feature_importance = min_feature_importance
+        self.max_num_categories = max_num_categories
+        self.sample_size = sample_size
+        self.random_state = random_state
+
+    def run(self, train_dataset, test_dataset, model=None) -> CheckResult:
+        """Run check.
+
+        Args:
+            train_dataset (Dataset): The training dataset object. Must contain a label column.
+            test_dataset (Dataset): The test dataset object. Must contain a label column.
+            model: not used in this check.
+
+        Returns:
+            CheckResult:
+                value: dictionary containing the domain classifier auc and a dict of column name to its feature
+                       importance as calculated for the domain classifier model.
+                display: distribution graph for each column for the columns most explaining the dataset difference,
+                         comparing the train and test distributions.
+
+        Raises:
+            DeepchecksValueError: If the object is not a Dataset or DataFrame instance
+        """
+
+        train_dataset = Dataset.validate_dataset_or_dataframe(train_dataset)
+        test_dataset = Dataset.validate_dataset_or_dataframe(test_dataset)
+
+        features = train_dataset.validate_shared_features(test_dataset, self.__class__.__name__)
+        cat_features = train_dataset.validate_shared_categorical_features(test_dataset, self.__class__.__name__)
+        self._cat_features = cat_features
+
+        model = self._generate_model(list(set(features) - set(cat_features)), cat_features)
+
+        sample_size = min(self.sample_size, train_dataset.n_samples, test_dataset.n_samples)
+        train_sample_df = train_dataset.features_columns.sample(sample_size, random_state=self.random_state)
+        test_sample_df = test_dataset.features_columns.sample(sample_size, random_state=self.random_state)
+
+        # create new dataset, with label denoting whether sample belongs to test dataset
+        domain_class_df = pd.concat([train_sample_df, test_sample_df])
+        domain_class_labels = pd.Series([0] * len(train_sample_df) + [1] * len(test_sample_df))
+
+        from sklearn.model_selection import train_test_split
+        x_train, x_test, y_train, y_test = train_test_split(domain_class_df, domain_class_labels,
+                                                            stratify=domain_class_labels,
+                                                            random_state=self.random_state,
+                                                            test_size=0.3)
```

Comment:
nitpicking and no one will use it, and yet - I think test_size should be a check parameter and not hard-coded here